src/vos/vos_aggregate.c

```diff
@@ -1537,24 +1564,30 @@ aggregate_exit(struct vos_container *cont, bool discard)
 }
 
 static void
-merge_window_init(struct agg_merge_window *mw)
+merge_window_init(struct agg_merge_window *mw, bool cont_has_csums)
 {
-	struct agg_io_context	*io = &mw->mw_io_ctxt;
+	struct vos_agg_io_context *io = &mw->mw_io_ctxt;
 
 	memset(mw, 0, sizeof(*mw));
 	D_INIT_LIST_HEAD(&mw->mw_phy_ents);
 	D_INIT_LIST_HEAD(&io->ic_nvme_exts);
+	mw->mw_csum_support = cont_has_csums;
+	io->ic_csum_buf = NULL;
+	io->ic_csum_buf_len = 0;
 }
 
 int
-vos_aggregate(daos_handle_t coh, daos_epoch_range_t *epr)
+vos_aggregate(daos_handle_t coh, daos_epoch_range_t *epr,
+	      unsigned int agg_flags)
```

Comment:
checksum is a container property, it should be got from container instead of specified by caller. I mean, if checksum is enabled for a container, we have to take care of checksums no matter what 'agg_flags' is specified, right?

Reply:
Agreed too.